docs/Custom_Device_Types.md

```diff
@@ -0,0 +1,184 @@
+
+The optional status notifier element is used to notify the engine of the last state of the custom device, and to indicate the device has completed execution. If this control is not used, a default No Error value is returned to the system when the device finishes execution. This error state is not checked until the system shuts down. Use an output channel to send more immediate status values to the system.
+
+The asynchronous framework includes VIs from the VeriStand Asynchronous Device Properties VIs subpalette.
+
+![](images/Asynchronous_Custom_Device_Framework.JPG)
+
+**Figure: Asynchronous Custom Device Framework**
+
+
+### Inline Hardware Interface
+
+The [inline hardware interface template](https://www.ni.com/documentation/en/veristand/latest/manual/custom-device-inline-hardware-interface-template/) is similar to a state machine architecture. Some developers will recognize it as an action-engine. See **[NI Discussion Forums](https://forums.ni.com/?profile.language=en)** » **[LabVIEW](https://forums.ni.com/t5/LabVIEW/bd-p/170?profile.language=en)** » **[Community Nugget 4/08/2007 Action Engines](https://forums.ni.com/t5/LabVIEW/Community-Nugget-4-08-2007-Action-Engines/m-p/503801?profile.language=en&requireLogin=False)** for a discussion on action engines. The PCL specifies the case to execute. An uninitialized **[Feedback Node](https://zone.ni.com/reference/en-XX/help/371361R-01/lvconcepts/block_diagram_feedback/)** is used for iterative data transfer. There are five cases defined by the Operation enumerated control.
+
+1.	Initialize
+2.	Start
+3.	Read Data from Hardware
+4.	Write Data to Hardware
+5.	Close
+
+This custom device runs in-line with the PCL, which calls each case at a specific time with respect to the other components in the VeriStand engine. The PCL will not proceed until the custom device case has completed.
+
+### Initialize
+
+The Initialize case executes before the PCL starts. In this case, you can read the device configuration information from properties using the reference to the device. Initialize data and buffers used internally in the device. The framework compiles the list of Data References for the custom device Inputs and Outputs in advance using **[Custom Device API](https://zone.ni.com/reference/en-XX/help/372846M-01/veristandmerge/vs_custom_device_api_vis_pal/)** » **[Driver Functions](https://zone.ni.com/reference/en-XX/help/372846M-01/veristandmerge/vs_driver_functions_vis_pal/)** » **[Get Custom Device Channel List](https://zone.ni.com/reference/en-XX/help/372846M-01/veristandmerge/vs_get_custom_device_channel_list_vi/)** and Custom Device API.lvlib » Templates » RT Driver VIs » Inline » Inline Driver Utilities » Channel Data References » Get Channel Data Reference.
```

Comment:
There are some issues with that path to Get Channel Data Reference.  There is help for that function.  Here is an updated chain of help doc links to lead to Get Channel Data Reference:

Custom Device API   https://zone.ni.com/reference/en-XX/help/372846M-01/veristandmerge/vs_custom_device_api_vis_pal/
>>
Driver Functions  https://zone.ni.com/reference/en-XX/help/372846M-01/veristandmerge/vs_driver_functions_vis_pal/
>>
Data References   https://zone.ni.com/reference/en-XX/help/372846M-01/veristandmerge/vs_data_references_pal/
>>
Get Channel Data Reference  https://zone.ni.com/reference/en-XX/help/372846M-01/veristandmerge/vs_get_channel_data_reference/